internal/download/get.go

```diff
@@ -39,66 +40,80 @@ func SetMocking(useMocking bool) {
 	}
 }
 
-func httpGet(url string) ([]byte, *failures.Failure) {
+func httpGet(url string) ([]byte, error) {
 	logging.Debug("Retrieving url: %s", url)
 	return httpGetWithProgress(url, nil)
 }
 
-func httpGetWithProgress(url string, progress *progress.Progress) ([]byte, *failures.Failure) {
-	logging.Debug("Retrieving url: %s", url)
-	client := retryhttp.NewClient(0 /* 0 = no timeout */, 5)
-	resp, err := client.Get(url)
-	if err != nil {
-		code := -1
-		if resp != nil {
-			code = resp.StatusCode
+func httpGetWithProgress(url string, progress *progress.Progress) ([]byte, error) {
+	var bs []byte
+	fn := func() error {
+		logging.Debug("Retrieving url: %s", url)
+		client := retryhttp.NewClientFromExisting(retryhttp.DefaultClient, 3)
+		resp, err := client.Get(url)
+		if err != nil {
+			code := -1
+			if resp != nil {
+				code = resp.StatusCode
+			}
+			fail := failures.FailNetwork.Wrap(err, locale.Tl("err_network_get", "Status code: {{.V0}}", strconv.Itoa(code)))
```

Comment:
You changed the function to return errors instead of failures, so return errors.